game-core/src/main/java/games/strategy/engine/chat/Chat.java

```diff
@@ -96,7 +98,7 @@ public void participantRemoved(final PlayerName playerName) {
   @Override
   public void slappedBy(final PlayerName from) {
     final String message = "You were slapped by " + from;
-    chatHistory.add(new ChatMessage(message, from));
+    chatHistory.add(new ChatMessage(from, message));
     chatMessageListeners.forEach(listener -> listener.slapped(message, from));
```

Comment:
Why is the new ChatMessage object not passed to the listener?

Reply:
There is a 'slapped' event object that is coming along that should be part of the slapped API. It's a bit weird though as slap is handled essentially as a special cased message and not its own event. TBD exactly how to best handle this.